ffmpeg/videoprofile.go

```diff
@@ -16,6 +16,16 @@ const (
 	FormatMP4
 )
 
+type Profile int
+
+const (
+	ProfileNone Profile = iota
+	H264Baseline
+	H264Main
+	H264High
+	H264ConstrainedHigh
+)
```

Comment:
Should we add some way to actually enumerate the profiles, similar to the `FormatExtensions` table? Maybe something like a `map[Profile]bool` . (There might be something else we can use as a value rather than a bool, but I can't think of anything.) 

That helps check whether a certain int (enum) supplied is actually a valid profile. go-livepeer uses these in a few places to fail out on missing support for new formats, eg

https://github.com/livepeer/go-livepeer/blob/ed7f24c763ad8db4592db5b4d1a2f2ce83aa6ca0/common/util_test.go#L181-L183

or

https://github.com/livepeer/go-livepeer/pull/1539/files#diff-3d8fe498e1aa211419d9b8f4b696b528R132-R136

For example, this could help check that the `Transcode` function within LPMS handles all profiles, or the webhook JSON conversion function in go-livepeer.

Reply:
Added a `map[Profile]string` that can also be used to find the profile parameters to pass to FFmpeg along with enumeration.